products/web_analytics/dags/cache_warming.py

```diff
@@ -0,0 +1,236 @@
+import json
+
+import dagster
+import posthoganalytics
+from dagster import Backoff, Jitter, RetryPolicy
+from prometheus_client import Counter, Gauge
+
+from posthog.hogql.constants import LimitContext
+
+from posthog.api.services.query import process_query_dict
+from posthog.clickhouse.client import sync_execute
+from posthog.clickhouse.query_tagging import Feature, tag_queries
+from posthog.dags.common import JobOwners
+from posthog.dags.common.resources import PostHogAnalyticsResource
+from posthog.exceptions_capture import capture_exception
+from posthog.hogql_queries.query_runner import ExecutionMode
+from posthog.models import Team
+from posthog.models.instance_setting import get_instance_setting
+
+from products.web_analytics.dags.web_preaggregated_utils import check_for_concurrent_runs
+
+STALE_WEB_QUERIES_GAUGE = Gauge(
+    "posthog_cache_warming_stale_web_query_gauge",
+    "Number of stale web queries present",
+    ["team_id"],
+)
+PRIORITY_WEB_QUERIES_COUNTER = Counter(
+    "posthog_cache_warming_priority_web_queries",
+    "Number of priority web queries warmed",
+    ["team_id", "normalized_query_hash", "is_cached"],
+)
+
+cache_warming_retry_policy = RetryPolicy(
+    max_retries=3,
+    delay=2,
+    backoff=Backoff.EXPONENTIAL,
+    jitter=Jitter.FULL,
+)
+
+
+def teams_enabled_for_web_analytics_cache_warming() -> list[int]:
```

Comment:
I still think this will be pretty slow and error prone (we can test it anyway). 

It would be great to rely on something else to pre-filter the query so we could at least filter `Team.objects` using the Django query builder before checking the FF every time.

For the time being, you could override this to allow a hardcoded list of teams so we can test the actual query warming behavior and optimize this selection later.